### PR TITLE
feat: add support for attaching custom policies

### PR DESCRIPTION
**Pull request by mtmn**
This PR adds support for attaching custom policies to the role same way that [iam-assumable-role-with-oidc](https://github.com/terraform-aws-modules/terraform-aws-iam/tree/v5.2.0/modules/iam-assumable-role-with-oidc) module does.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -64,3 +64,15 @@ variable "role_permissions_boundary" {
...
   default     = null
 }
+
+variable "role_policy_arns" {
+  description = "List of ARNs of IAM policies to attach to IAM role"
+  type        = list(string)
+  default     = []
+}
+
+variable "number_of_role_policy_arns" {
```
Comment (npalm) [inline variables.tf:74]: What is the reason to overwrite the number of policies?

**Review by npalm — COMMENTED** [review]
Thansk for creating the PR, sorry for the late reply.

File: variables.tf (hunk 1 above)
Comment (mtmn) [reply variables.tf:74]: to allow attachment of multiple policies --


e.g.
```
  role_policy_arns = [
    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryPowerUser",
    dependency.policies.outputs.policy_arns.test
  ]
```

Comment (npalm) [reply variables.tf:74]: But the count can be calculated based on the input of the variable `role_policy_arns` So the count looks not needed.

Comment (mtmn) [reply variables.tf]: good point, removed.

**Review by npalm — APPROVED** [review]

**Pull request merged**